controller/rpc/scanner.go

```diff
@@ -3,193 +3,242 @@ package rpc
-		creditPool: make(chan struct{}, scannerLBMax*(maxConns+1)),
-	}
+// ScannerAcquisitionManager is responsible for managing the scan credit for the scanners.
+type ScannerAcquisitionManager struct {
+	maxConcurrentScansPerScanner int
+	scannerHealthChecker         ScannerHealthCheckFunc
+	clusterHelper                kv.ClusterHelper
+	requestChan                  chan *acquireScannerRequest
+	workerStarted                sync.Once
 }
 
-func (mgr *ScanCreditManager) decScanningCount(sid string) {
-	mgr.mutex.Lock()
-	defer mgr.mutex.Unlock()
+var ScannerAcquisitionMgr *ScannerAcquisitionManager
 
-	err := mgr.scannerLoadBalancer.ReleaseScanCredit(sid)
-	if err != nil {
-		log.WithFields(log.Fields{"error": err, "scanner": sid}).Error()
+const (
+	JitterFactor = 0.2
+	BaseDelay    = time.Millisecond * 500
+	MaxDelay     = time.Second * 5
+)
+
+// ScannerHealthCheckFunc defines the signature for a scanner health check function
+// It pings a scanner to verify it's available before assigning scan tasks to it
+type ScannerHealthCheckFunc func(scannerID string, timeout time.Duration) error
+
+func NewScannerAcquisitionManager(maxConcurrentScansPerScanner, maxConcurrentRepoScanTasks int) *ScannerAcquisitionManager {
+	mgr := &ScannerAcquisitionManager{
+		maxConcurrentScansPerScanner: maxConcurrentScansPerScanner,
+		clusterHelper:                kv.GetClusterHelper(),
+		requestChan:                  make(chan *acquireScannerRequest, maxConcurrentRepoScanTasks*2), // Buffer for up to maxConcurrentRepoScanTasks * 2 pending requests,
 	}
+	mgr.SetScannerHealthChecker(mgr.Ping)
+	return mgr
 }
 
-func (mgr *ScanCreditManager) releaseScanCredit(sid string) {
-	mgr.mutex.Lock()
-	defer mgr.mutex.Unlock()
+func (mgr *ScannerAcquisitionManager) SetScannerHealthChecker(healthChecker ScannerHealthCheckFunc) {
+	mgr.scannerHealthChecker = healthChecker
+}
 
-	err := mgr.scannerLoadBalancer.ReleaseScanCredit(sid)
-	if err != nil {
-		log.WithFields(log.Fields{"error": err, "scanner": sid}).Error()
-	} else {
-		mgr.signalScanCredit()
+func (mgr *ScannerAcquisitionManager) GetMaxConcurrentScansPerScanner() int {
+	return mgr.maxConcurrentScansPerScanner
+}
+
+func (mgr *ScannerAcquisitionManager) SetClusterHelper(clusterHelper kv.ClusterHelper) {
+	mgr.clusterHelper = clusterHelper
+}
+
+func (mgr *ScannerAcquisitionManager) GetClusterHelper() kv.ClusterHelper {
+	return mgr.clusterHelper
+}
+
+// backoffDelay calculates exponential backoff with jitter
+func (mgr *ScannerAcquisitionManager) backoffDelay(attempts int) time.Duration {
+	if attempts <= 1 {
+		return BaseDelay
+	}
+
+	exp := float64(BaseDelay) * math.Pow(2, float64(attempts-1))
+	if exp > float64(MaxDelay) {
+		exp = float64(MaxDelay)
 	}
+
+	// between [1-JitterFactor, 1+JitterFactor]
+	jitter := 1 - JitterFactor + rand.Float64()*(2*JitterFactor)
+	delay := exp * jitter
+
+	return time.Duration(delay)
+}
```

Comment:
Can we use an existing package to do the exponential retry?  For example, your backoffDelay() can be replaced with [Backoff.Step()](https://pkg.go.dev/k8s.io/apimachinery/pkg/util/wait#Backoff.Step).